Zend/zend_object_handlers.c

```diff
@@ -1541,6 +1541,10 @@ ZEND_API zend_function *zend_std_get_static_method(zend_class_entry *ce, zend_st
 	if (EXPECTED(fbc)) {
 		if (UNEXPECTED(fbc->common.fn_flags & ZEND_ACC_ABSTRACT)) {
 			zend_abstract_method_call(fbc);
+			if (UNEXPECTED(fbc->common.fn_flags & ZEND_ACC_CALL_VIA_TRAMPOLINE)) {
+				zend_string_release_ex(fbc->common.function_name, 0);
+				zend_free_trampoline(fbc);
```

Comment:
Unrelated, but I think we need this below in the ZEND_ACC_TRAIT case when EG(exception) is set.

Reply:
Indeed:

```
php: php-src/Zend/zend_execute_API.c:489: void shutdown_executor(void): Assertion `(executor_globals.trampoline).common.function_name == ((void*)0) || (compiler_globals.unclean_shutdown)' failed.
```

I'll make a separate PR for that.

Reply:
#17728.